### PR TITLE
feat(validation): Validation package

### PR DESCRIPTION
**Pull request by sgangnant**
**What is the problem this PR is trying to solve?**
Several projects in Talend have the same validation requirements on specific fields (first name, last name, email ...).
Checks are repeated across projects and are not harmonized.

**What is the chosen solution to this problem?**
Introduce a package that exposes helpers to check specific values

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/validation/src/methods.test.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import { validFirstName, validLastName, validEmail, validPhone } from './methods';
```
Comment (sgangnant) [inline packages/validation/src/methods.test.ts:1]: Test cases clearly show what's acceptable and what's not

File: packages/validation/src/methods.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { NAME, EMAIL, DOMAIN, PHONE } from './regexp';
+
+/**
+ * Check that a given value is a valid first name
+ * @param {String} value
+ * @returns {Boolean}
+ */
+export function validFirstName(value: string): Boolean {
```
Comment (gjou-tlnd) [inline packages/validation/src/methods.ts:8]: what is the reason for using a Boolean instead of the primitive type?

**Review by tlnd-mhuchet — APPROVED** [review]
Looks nice 👍

File: packages/validation/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# @talend/validation
+
+This packages exposes utils to validate values against specific shapes they should follow (names, emails ...).
+
+## Getting started
+
+Add the library to the dependencies of your project
+
+```javascript
```
Comment (frassinier) [inline packages/validation/README.md:9]:
```suggestion
```sh
```

File: packages/validation/README.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+npm i @talend/validation
+```
+
+or
+
+```javascript
```
Comment (frassinier) [inline packages/validation/README.md:15]:
```suggestion
```sh
```

File: packages/validation/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+yarn add @talend/validation
+```
+
+Then use the exposed methods/regular expression exposed by the library.
```
Comment (frassinier) [inline packages/validation/README.md:19]:
```suggestion
Then use the methods/regular expressions exposed by the library.
```

File: packages/validation/src/methods.test.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+			(value: string) => expect(validEmail(value)).toBe(false),
+		);
+	});
+
+	describe('validPhone', () => {
+		const validPhones: Array<string> = [
+			'+33102030405',
+		];
+
+		const invalidPhones: Array<string> = [
```
Comment (frassinier) [inline packages/validation/src/methods.test.ts:99]: Could you extract them into new files for more scalability. I mean, all the examples

**Review by frassinier — APPROVED** [review]

**Pull request merged**